internal/server/iigointernal/orchestration.go

```diff
@@ -76,64 +91,86 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
 	executiveBranch.loadClientPresident(presidentPointer)
 	legislativeBranch.loadClientSpeaker(speakerPointer)
 
-	// Withdraw the salaries
-	presidentWithdrawSuccess := judicialBranch.withdrawPresidentSalary(g)
-	judgeWithdrawSuccess := legislativeBranch.withdrawJudgeSalary(g)
-	speakerWithdrawSuccess := executiveBranch.withdrawSpeakerSalary(g)
+	// Pay salaries into budgets
 
-	// Handle the lack of resources
-	if !presidentWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since President has no resources to spend"
-	}
-	if !judgeWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since Judge has no resources to spend"
-	}
-	if !speakerWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since Speaker has no resources to spend"
+	errorJudicial := judicialBranch.sendPresidentSalary()
+	errorLegislative := legislativeBranch.sendJudgeSalary()
+	errorExecutive := executiveBranch.sendSpeakerSalary()
```

Comment:
can you use the `err != nil` pattern?

Reply:
In the current implementation, we want to pay as many roles as possible. We therefore try paying all three roles and then return an error if that did not succeed.  Are you proposing that we check if a call returns nil before running the next function call?

Reply:
> Are you proposing that we check if a call returns nil before running the next function call?
Yeah

Ok then. can you add this as a comment because this is not the usual error handling behaviour?

Reply:
Done